runners/core-java/src/main/java/org/apache/beam/runners/core/SplittableProcessElementInvoker.java

```diff
@@ -51,11 +51,6 @@ public Result(
         @Nullable WatermarkEstimatorStateT futureWatermarkEstimatorState) {
       checkArgument(continuation != null, "continuation must not be null");
       this.continuation = continuation;
-      if (continuation.shouldResume()) {
-        checkArgument(
-            residualRestriction != null,
-            "residual restriction must not be null if continuation indicate it should resume");
-      }
       this.residualRestriction = residualRestriction;
       this.futureOutputWatermark = futureOutputWatermark;
       this.futureWatermarkEstimatorState = futureWatermarkEstimatorState;
```

Comment:
I believe the comment below could be incorrect. If we get stop(), we shouldn't have a residual restriction.